url11fb0973cf_raichuman9_Swimmer526_tgz-pJ8-fr_inria_optimization_cmaes_examples_CMAExample1J8/src/org/jbox2d/dynamics/contacts/ContactSolver.java

```diff
@@ -0,0 +1,908 @@
+/*******************************************************************************
+ * Copyright (c) 2011, Daniel Murphy
+ * All rights reserved.
+ * 
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ * 	* Redistributions of source code must retain the above copyright notice,
+ * 	  this list of conditions and the following disclaimer.
+ * 	* Redistributions in binary form must reproduce the above copyright notice,
+ * 	  this list of conditions and the following disclaimer in the documentation
+ * 	  and/or other materials provided with the distribution.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ ******************************************************************************/
+package org.jbox2d.dynamics.contacts;
+
+import org.jbox2d.collision.Manifold;
+import org.jbox2d.collision.ManifoldPoint;
+import org.jbox2d.collision.WorldManifold;
+import org.jbox2d.collision.shapes.Shape;
+import org.jbox2d.common.Mat22;
+import org.jbox2d.common.MathUtils;
+import org.jbox2d.common.Settings;
+import org.jbox2d.common.Vec2;
+import org.jbox2d.dynamics.Body;
+import org.jbox2d.dynamics.Fixture;
+
+// updated to rev 100
+// pooled locally, non-threaded
+/**
+ * @author Daniel
+ */
+public class ContactSolver {
+	
+	/**
+	 * For each solver, this is the initial number of constraints in the array, which expands
+	 * as needed.
+	 */
+	public static final int INITIAL_NUM_CONSTRAINTS = 256;
+	
+	/**
+	 * Ensure a reasonable condition number. for the block solver
+	 */
+	public static final float k_maxConditionNumber = 100.0f;
+	
+	public ContactConstraint[] m_constraints;
+	public int m_constraintCount;
+	
+	public ContactSolver(){
+		m_constraints = new ContactConstraint[INITIAL_NUM_CONSTRAINTS];
+		for(int i=0; i< m_constraints.length; i++){
+			m_constraints[i] = new ContactConstraint();
+		}
+	}
+	
+	// djm pooling
+	private final WorldManifold worldManifold = new WorldManifold();
+	private final Vec2 tangent = new Vec2();
+	private final Vec2 temp1 = new Vec2();
+	private final Vec2 temp2 = new Vec2();
+	
+	public final void init(Contact[] contacts, int contactCount, float impulseRatio){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 97 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.